app/src/main/scala/org/alephium/explorer/cache/BlockCache.scala

```diff
@@ -95,12 +96,9 @@ object BlockCache {
           .buildAsync[ChainIndex, Duration](blockTimeAsyncLoader)
       }
 
-    val cacheRowCount: CaffeineAsyncCache[Query[_, _, Seq], Int] =
-      CaffeineAsyncCache.rowCountCache(DBRunner(dc)) {
-        Caffeine
-          .newBuilder()
-          .refreshAfterWrite(5, java.util.concurrent.TimeUnit.SECONDS)
-      }
+    val cacheRowCount: AsyncReloadingCache[Int] = AsyncReloadingCache(0, 10.seconds) { _ =>
+      run(BlockQueries.mainChainQuery.length.result)
+    }
```

Comment:
@tdroxler this cache is loading synchronously, so for the first 10.seconds `0` will be returned the clients. We added `AsyncReloadingCache.reloadNow` for this use-case so we get guarantee that the cache is populated before the boot-up is complete. 

But it's not being used so I'm assuming this how you'd want #151 resolved as well? With `Seq.empty[BlockEntryLite]` as cache value for the first n seconds?

Reply:
mm yeah it would be better to use `reloadNow`. What I did in the `TransationCache` was [to extends it as a Service and invalidate the cache in the startSelfOnce function](https://github.com/alephium/explorer-backend/blob/3d3c9bfb9f8060f0488ca3129d65a59807bcf300/app/src/main/scala/org/alephium/explorer/cache/TransactionCache.scala#L45-L60), maybe we could use `reloadNow` instead? but anyway we could also add `BlockCache extends Service` so we wait that the block cache is populated to start the app.

Reply:
To avoid the same story again. Probably better to leave this to you to however you like/prefer to use `reloadNow` with `Service`.